### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by NikitaKarbazin**

### CODE REVIEW

**Review by Kaunin1 — APPROVED** [review]
Well done !

File: src/herbivoresAndCarnivores.js [hunk 1]
```diff
@@ -1,15 +1,33 @@
...
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor() {
+    super();
+    this.hidden = false;
+  }
+  hide() {
+    this.hidden = !this.hidden;
+  }
```
Comment (jstmpelowycz) [inline src/herbivoresAndCarnivores.js:20]:
```suggestion
  constructor() {
    super();
    this.hidden = false;
  }

  hide() {
    this.hidden = !this.hidden;
  }
```

File: src/herbivoresAndCarnivores.js [hunk 2]
```diff
@@ -1,15 +1,33 @@
...
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(herbivore) {
+    if (!herbivore.hidden && herbivore instanceof Herbivore) {
+      herbivore.health -= 50;
+    }
+
+    Animal.alive = Animal.alive.filter(animal => animal.health);
```
Comment (jstmpelowycz) [inline src/herbivoresAndCarnivores.js:29]: it works, however, every time `bite` function is called, this filtering gets performed anyway, do you think it's okay?

Comment (NikitaKarbazin) [reply src/herbivoresAndCarnivores.js]: Agree with u! I wanted to do more simple code! But not suppose the main logic! Fixed!

**Review by illiarerun — APPROVED** [review]

**Pull request closed without merging**